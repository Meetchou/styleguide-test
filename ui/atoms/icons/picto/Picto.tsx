import React, { FC, MouseEvent } from 'react';
import cls from 'classnames';

import { IconType } from '@ui/types';
import { getIcon } from '@ui/atoms/icons/icon/utils';

export type PictoType = {
  icon?: IconType,
  onIconClick?: (event: MouseEvent) => void,
  className?: string
};


const defaultClassNames = [
  'text-sm',
  'text-gray-400',
  'hover:text-primary-500',
  'transition-colors',
  'cursor-pointer',
  'text-center'
];

const Picto: FC<PictoType> = ({ icon, onIconClick, className }) => {
  const Icon: React.ComponentType | undefined = getIcon(icon);
  if (!Icon) {
    return null;
  }


  if (!onIconClick) {
    return (
      <span className={className}>
        <Icon />
      </span>
    );
  }

  return (
    <span
      className={cls(defaultClassNames, className)}
      onClick={onIconClick}
      aria-hidden='true'
    >
      <Icon />
    </span>
  );
};

export default Picto;
