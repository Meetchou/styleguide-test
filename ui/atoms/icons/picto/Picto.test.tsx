import React from 'react';

import render from '@ui/testing/render';

import Picto from './Picto';

describe('atoms/icons/picto', () => {
  describe('renders correctly, with checked value:', () => {
    it('render correctly', () => {
      const tree = render(<Picto icon='home' />);
      expect(tree.toJSON()).toMatchSnapshot();
    });

    it('classname props is correctly passed', () => {
      const tree = render(<Picto icon='home' className='bg-primary-500' />);
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
