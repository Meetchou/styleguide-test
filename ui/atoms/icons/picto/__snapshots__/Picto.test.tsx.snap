// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/icons/picto renders correctly, with checked value: classname props is correctly passed 1`] = `
<span
  className="bg-primary-500"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
    />
  </svg>
</span>
`;

exports[`atoms/icons/picto renders correctly, with checked value: render correctly 1`] = `
<span>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
    />
  </svg>
</span>
`;
